maven-extension/src/main/java/io/opentelemetry/maven/OpenTelemetrySdkService.java

```diff
@@ -77,10 +78,12 @@ public synchronized void dispose() {
                 + sdkProviderShutdown.isDone()
                 + ")");
       }
-      // reset GlobalOpenTelemetry as we have just shutdown its TraceProvider
-      GlobalOpenTelemetry.resetForTest();
       this.openTelemetrySdk = null;
     }
+    // reset GlobalOpenTelemetry as we have just shutdown its TraceProvider
+    GlobalOpenTelemetry.resetForTest();
```

Comment:
I would also add a comment about this being a hack, since this is definitely not intended for anything exception unit testing purposes. It's one of the reasons that I strongly recommend avoiding using the Global instance if you possibly can.

Reply:
Good callout - I guess everything should still work even if we remove `GlobalOpenTelemetry.set` and use `build()` instead of `buildAndRegisterGlobal` below?

Reply:
I was interested in enabling `GlobalOpenTelemetry` as the static helper methods are commonly used when using Otel in Java and I was influenced by the fact that the AutoConfigure module was relying on `buildAndRegisterGlobal`.

I'm happy to move away from `GlobalOpenTelemetry`, it should be straightforward as the usage of the Otel APIs in Maven are limited, for the moment, to the code of the OpenTelemetry Maven Extension. Maybe we will have to evolve when we want to use the Otel APIs in other Maven mojos.

Reply:
Is the preference to remove use of the Global OTeL instance completely and just rely on the local one?

Reply:
@kenfinnigan in fact, we are not using GlobalOpenTelemetry at the moment, the primary reason being that I was interested in standardising the name of the tracer so the [OtelExecutionListener](https://github.com/open-telemetry/opentelemetry-java-contrib/blob/v1.6.0/maven-extension/src/main/java/io/opentelemetry/maven/OtelExecutionListener.java) just relies on [OpenTelemetrySdkService#getTracer()](https://github.com/open-telemetry/opentelemetry-java-contrib/blob/v1.6.0/maven-extension/src/main/java/io/opentelemetry/maven/OpenTelemetrySdkService.java).

I'm confident that removing the `registerGlobal` part will have no impact on the existing code.

Does this clarify?

Reply:
Sure, will make the change

Reply:
@cyrille-leclerc I've pushed a change to remove the use of `GlobalOpenTelemetry`